tds.go

```diff
@@ -835,23 +1021,54 @@ initiate_connection:
 	}
 
 	login := login{
-		TDSVersion:   verTDS74,
-		PacketSize:   uint32(outbuf.PackageSize()),
-		Database:     p.database,
-		OptionFlags2: fODBC, // to get unlimited TEXTSIZE
-		HostName:     p.workstation,
-		ServerName:   p.host,
-		AppName:      p.appname,
-		TypeFlags:    p.typeFlags,
+		TDSVersion:          verTDS74,
+		PacketSize:          uint32(outbuf.PackageSize()),
+		Database:            p.database,
+		OptionFlags2:        fODBC, // to get unlimited TEXTSIZE
+		HostName:            p.workstation,
+		ServerName:          p.host,
+		AppName:             p.appname,
+		TypeFlags:           p.typeFlags,
+		FedAuthLibrary:      p.fedAuthLibrary,
+		FedAuthEcho:         fedAuthEcho,
+		FedAuthADALWorkflow: p.fedAuthADALWorkflow,
 	}
 	auth, auth_ok := getAuth(p.user, p.password, p.serverSPN, p.workstation)
-	if auth_ok {
+	if login.FedAuthLibrary == fedAuthLibraryReserved && auth_ok {
```

Comment:
This section +30 lines of the nested else/if needs to be cleaned up. using switch may help, and maybe try to get rid of nested ifs with early returns.

Reply:
I have restructured this as a switch.